apps/nextjs-website/src/components/organisms/Auth/EmailFormWrapper.tsx

```diff
@@ -0,0 +1,32 @@
+import EditEmailForm from '@/components/organisms/Auth/EditEmailForm';
+import {
+  InfoCardItem,
+  InfoCardItemProps,
+} from '@/components/atoms/InfoCardItem/InfoCardItem';
+import React from 'react';
+
+type EmailFormWrapperProps = {
+  item: InfoCardItemProps;
+  isEditing: boolean;
+  // eslint-disable-next-line functional/no-return-void
+  onCancel: () => void;
+  onSave: (newEmail: string) => Promise<void>;
+  // eslint-disable-next-line functional/no-return-void
+  onEdit: () => void;
+};
+
+const EmailFormWrapper: React.FC<EmailFormWrapperProps> = ({
+  item,
+  isEditing,
+  onCancel,
+  onSave,
+  onEdit,
+}) => {
+  return isEditing ? (
+    <EditEmailForm onCancel={onCancel} onSave={onSave} />
+  ) : (
+    <InfoCardItem {...item} onEdit={onEdit} />
+  );
+};
```

Comment:
IMHO this could be a good example to start with unit testing on components: the `EmailFormWrapper` behaves differently based on a props, so I'd make sure to check the proper component is rendered.

What do you think?

Reply:
I think you're right. I split this activity to [this separate subtask](https://pagopa.atlassian.net/browse/DEV-1314) to keep small the PR size